entrypoint.sh

```diff
@@ -13,12 +16,81 @@ request_body=$(cat <<EOF
 EOF
 )
 
-echo "Request_body: $request_body"
+  echo "API URL: $deployment_trigger_url"
+  echo "Request Body: $request_body"
+
+  STATUS_CODE=$(curl -sSk  -o response_body.txt -w "%{http_code}" -X POST "$deployment_trigger_url" \
+    -H "Content-Type: application/json" \
+    -H "x-api-key: $INPUT_KAPSTAN_API_KEY" \
+    -d "$request_body")
+  
+  echo "Response Body: $(cat response_body.txt)"
+  echo "Response Status Code: $STATUS_CODE"
+  DEPLOYMENT_ID=$(cat response_body.txt | jq -r '.deployment_id')
+  
+  if [[ -z "$DEPLOYMENT_ID"  || $STATUS_CODE != 2* ]];
+  then
+    echo "Failed to deploy app, err: $(cat response_body.txt)"
+    exit 1
+  fi
+  echo "KAPSTAN_DEPLOYMENT_ID=$DEPLOYMENT_ID" >> "${GITHUB_ENV}"
```

Comment:
if we exit here, then the file response_body.txt will not get removed.

Reply:
Thanks for pointing out. Done, added `trap` to take responsibility of cleanup on EXIT